feature/exchange-gain-loss (#215)
fivetran/dbt_quickbooks: CHANGELOG.md

# dbt_quickbooks v1.9.0-a1
[PR #212](https://github.com/fivetran/dbt_quickbooks/pull/212) includes the following updates:

## Schema/Data Change
**2 total changes • 0 possible breaking changes**

| Data Model(s) | Change type | Old | New | Notes |
| ------------- | ----------- | --- | --- | ----- |
| `quickbooks__general_ledger`<br>`quickbooks__general_ledger_by_period` | New records | — | Exchange gain/loss entries | |

## Feature Update

> - Adds realized exchange gain/loss entries to `int_quickbooks__bill_payment_double_entry` and `int_quickbooks__payment_double_entry` to capture the currency impact when a foreign currency transaction is settled at a different exchange rate than when it was originally recorded. Requires an `ExchangeGainOrLoss` account in QuickBooks to generate entries.

Reviewer: In the validation, it appears to be both an `account_name` value and an `account_sub_type` value. Do we need to call that out explicitly?

Author: Good question. It's the `account_sub_type` that is a pre-set list coming from Quickbook, and the name should follow, so yes I should clarify what field we're talking about here.

Author: Updated!

> - Adds realized exchange gain/loss entries to `int_quickbooks__bill_payment_double_entry` and `int_quickbooks__payment_double_entry` to capture the currency impact when a foreign currency transaction is settled at a different exchange rate than when it was originally recorded. Requires an `ExchangeGainOrLoss` account in QuickBooks to generate entries.

Author: ```suggestion
- Adds realized exchange gain/loss entries to `int_quickbooks__bill_payment_double_entry` and `int_quickbooks__payment_double_entry` to capture the currency impact when a foreign currency transaction is settled at a different exchange rate than when it was originally recorded. Requires an account with a subtype of `ExchangeGainOrLoss` in QuickBooks to generate entries.
```

# dbt_quickbooks v1.9.0

> [PR #212](https://github.com/fivetran/dbt_quickbooks/pull/212) includes the following updates:

Reviewer 2: ```suggestion
[PR #215](https://github.com/fivetran/dbt_quickbooks/pull/215) includes the following updates:
```

## Schema/Data Change
**2 total changes • 0 possible breaking changes**

| Data Model(s) | Change type | Old | New | Notes |
| ------------- | ----------- | --- | --- | ----- |

> | `quickbooks__general_ledger`<br>`quickbooks__general_ledger_by_period` | New records | — | Exchange gain/loss entries | |

Reviewer 2: We should remove this since by default there's no schema/data change. The Feature Update section can highlight this.

Author: Done.

## Feature Update

> - Adds realized exchange gain/loss entries to `int_quickbooks__bill_payment_double_entry` and `int_quickbooks__payment_double_entry` to capture the currency impact when a foreign currency transaction is settled at a different exchange rate than when it was originally recorded. Requires an account with a subtype of `ExchangeGainOrLoss` in QuickBooks to generate entries.

Reviewer 2: ```suggestion
- Adds the ability to realize exchange gain/loss entries to `int_quickbooks__bill_payment_double_entry` and `int_quickbooks__payment_double_entry`, which populate in `quickbooks__general_ledger` and quickbooks__general_ledger_by_period`, to capture the currency impact when a foreign currency transaction is settled at a different exchange rate than when it was originally recorded. Requires an account with a subtype of `ExchangeGainOrLoss` in QuickBooks to generate entries.
```

> - Adds the `using_exchange_gain_loss` variable (default `true`) to disable exchange gain/loss entry generation for users without multi-currency transactions or an Exchange Gain or Loss account. See the [README](https://github.com/fivetran/dbt_quickbooks/tree/main#disabling-exchange-gainloss-entries) for details.

Reviewer 2: Let's not set this a `true` by default to start. This is a major new accounting shift. While I agree this is necessary for users leveraging multi-currency and have an `ExchangeGainOrLoss` account, this may not be relevant for all users. 

Let's set this as `false` by default and highlight the reason you would enable this and offer the configuration for those who need it or want to try it. We can plan to swap this to `true` in a future release once we have full confidence in it's behavior.

Reviewer 2: Given this approach, we'll need to update the various files where we set this as `true` by default and where we document the default behavior.
